x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/querydsl/container/FieldExtractorRegistry.java

```diff
@@ -24,14 +24,14 @@
 public class FieldExtractorRegistry {
 
     private final Map<String, FieldExtraction> cache = new HashMap<>();
-    
+
     public FieldExtraction fieldExtraction(Expression expression) {
         return cache.computeIfAbsent(Expressions.id(expression), k -> createFieldExtractionFor(expression));
     }
 
     private FieldExtraction createFieldExtractionFor(Expression expression) {
         if (expression instanceof FieldAttribute) {
-            FieldAttribute fa = (FieldAttribute) expression;
+            FieldAttribute fa = ((FieldAttribute) expression).exactAttribute();
```

Comment:
This is a subtle one that otherwise a lot of things don't work. For example grouping a text field should only occur on a keyword.